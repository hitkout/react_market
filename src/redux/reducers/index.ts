import { combineReducers } from 'redux';

import productsReducer from './products';
import basketReducer from './basket';
import advantagesReducer from './advantages';
import productPageReducer from './productPage';

const rootReducer = combineReducers({
    products: productsReducer,
    basket: basketReducer,
    advantages: advantagesReducer,
    productPage: productPageReducer
})

export default rootReducer;