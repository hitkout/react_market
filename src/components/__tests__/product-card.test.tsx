import React from 'react';
import { mount } from 'enzyme';
import ProductCard from '../product-card';
import { Provider } from 'react-redux';
import { store } from '../../redux/store';
import { StaticRouter } from 'react-router-dom';
import { describe, it, expect } from '@jest/globals';

describe('Тестирование ProductCard', () => {

    it('Тестируем рендер ProductCard', async () => {
        const component = mount(
            <Provider store={store}>
                <StaticRouter>
                    <ProductCard id image name price onClickAddProducts addedCount processor motherboard videocard RAMmemory hdd hddMemory ssd powerSupplyUnit />
                </StaticRouter>
            </Provider>
        )

        expect(component).toMatchSnapshot()
    });
});