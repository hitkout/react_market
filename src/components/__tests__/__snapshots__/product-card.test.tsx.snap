// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование ProductCard Тестируем рендер ProductCard 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <StaticRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      staticContext={Object {}}
    >
      <productCard
        RAMmemory={true}
        addedCount={true}
        hdd={true}
        hddMemory={true}
        id={true}
        image={true}
        motherboard={true}
        name={true}
        onClickAddProducts={true}
        powerSupplyUnit={true}
        price={true}
        processor={true}
        ssd={true}
        videocard={true}
      >
        <h1
          className="productPageTitle"
        >
          <Component />
        </h1>
        <div
          className="containerRow"
        >
          <div
            className="productPageImage"
          >
            <img
              alt=""
            />
          </div>
          <div
            className="productPageContent"
          >
            <div
              className="productPagePrice"
            >
              <i>
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        384,
                        512,
                        Array [],
                        "f158",
                        "M239.36 320C324.48 320 384 260.542 384 175.071S324.48 32 239.36 32H76c-6.627 0-12 5.373-12 12v206.632H12c-6.627 0-12 5.373-12 12V308c0 6.627 5.373 12 12 12h52v32H12c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h52v52c0 6.627 5.373 12 12 12h58.56c6.627 0 12-5.373 12-12v-52H308c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12H146.56v-32h92.8zm-92.8-219.252h78.72c46.72 0 74.88 29.11 74.88 74.323 0 45.832-28.16 75.561-76.16 75.561h-77.44V100.748z",
                      ],
                      "iconName": "ruble-sign",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-ruble-sign fa-w-12 "
                    data-icon="ruble-sign"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 384 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M239.36 320C324.48 320 384 260.542 384 175.071S324.48 32 239.36 32H76c-6.627 0-12 5.373-12 12v206.632H12c-6.627 0-12 5.373-12 12V308c0 6.627 5.373 12 12 12h52v32H12c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h52v52c0 6.627 5.373 12 12 12h58.56c6.627 0 12-5.373 12-12v-52H308c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12H146.56v-32h92.8zm-92.8-219.252h78.72c46.72 0 74.88 29.11 74.88 74.323 0 45.832-28.16 75.561-76.16 75.561h-77.44V100.748z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </i>
            </div>
            <div>
              <button
                className="productPageButton"
                onClick={[Function]}
              >
                <i
                  className="addedCount"
                >
                  <Component />
                </i>
              </button>
            </div>
          </div>
        </div>
        <h2
          className="productPageDescription"
        />
        <p>
           
           
           
           
           
           
           
           
        </p>
        <h2
          className="productPageSpecificationsTitle"
        />
        <table
          className="productPageSpecifications"
        >
          <tbody>
            <tr>
              <td
                className="specName"
              />
              <td
                className="specValue"
              >
                <Component />
              </td>
            </tr>
            <tr>
              <td
                className="specName"
              />
              <td
                className="specValue"
              >
                <Component />
              </td>
            </tr>
            <tr>
              <td
                className="specName"
              />
              <td
                className="specValue"
              >
                <Component />
              </td>
            </tr>
            <tr>
              <td
                className="specName"
              />
              <td
                className="specValue"
              >
                <Component />
              </td>
            </tr>
            <tr>
              <td
                className="specName"
              />
              <td
                className="specValue"
              >
                 
              </td>
            </tr>
            <tr>
              <td
                className="specName"
              />
              <td
                className="specValue"
              >
                <Component />
              </td>
            </tr>
            <tr>
              <td
                className="specName"
              />
              <td
                className="specValue"
              >
                <Component />
              </td>
            </tr>
          </tbody>
        </table>
      </productCard>
    </Router>
  </StaticRouter>
</Provider>
`;
